src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -292,6 +292,51 @@ static bool M<T>(T x = default)
             Assert.Equal(Conversion.Identity, model.GetConversion(default2));
         }
 
+        [Fact, WorkItem(38643, "https://github.com/dotnet/roslyn/issues/38643")]
+        public void ComparisonWithGenericType_ValueType()
+        {
+            string source = @"
+class C
+{
+    static bool M<T>(T x = default) where T : struct
+    {
+        return x == default // 1
+            && x == default(T); // 2
+    }
+}
+";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics(
+                // (6,16): error CS0019: Operator '==' cannot be applied to operands of type 'T' and 'default'
+                //         return x == default // 1
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == default").WithArguments("==", "T", "default").WithLocation(6, 16),
+                // (7,16): error CS0019: Operator '==' cannot be applied to operands of type 'T' and 'T'
+                //             && x == default(T); // 2
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == default(T)").WithArguments("==", "T", "T").WithLocation(7, 16)
+                );
+        }
+
+        [Fact, WorkItem(38643, "https://github.com/dotnet/roslyn/issues/38643")]
+        public void ComparisonWithGenericType_ReferenceType()
+        {
+            string source = @"
+class C
+{
+    static bool M<T>(T x = default) where T : class
+    {
+        return x == default
+            && default == x
+            && x == default(T)
+            && default(T) == x;
+    }
+}
+";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics();
```

Comment:
>comp.VerifyDiagnostics(); [](start = 12, length = 25)

For success cases, I think we would want to run the code and observe that they behaves per spec.